### ISSUE TITLE
handshake json

### ISSUE THREAD

**Issue by johnshiver** [andybeers/KingKai#7]
{'HandshakeSecret': 'john_wuz_here',
 'HostInfo': {'bootTime': 1491668443,
              'hostid': '87c9fbe9-c8c8-4130-a161-04de4ebf52dd',
              'hostname': 'borg-3',
              'kernelVersion': '4.4.0-53-generic',
              'os': 'linux',
              'platform': 'ubuntu',
              'platformFamily': 'debian',
              'platformVersion': '16.04',
              'procs': 391,
              'uptime': 991360,
              'virtualizationRole': 'host',
              'virtualizationSystem': 'kvm'}}

**Comment by johnshiver** [comment]
eventually HandshakeSecret will be handshake_secret and HostInfo will be host_info I just cant figure out whats wrong with those values

**Comment by johnshiver** [comment]
payload with id


{"cpus":[{"cpu":"cpu0","user":15626.51,"system":27748.71,"idle":946910.49,"nice":30.92,"iowait":2880.95,"irq":0,"softirq":53.67,"steal":0,"guest":0,"guestNice":0,"stolen":0},{"cpu":"cpu1","user":15206.22,"system":27859.56,"idle":943744.05,"nice":63.1,"iowait":6302.53,"irq":0,"softirq":31.68,"steal":0,"guest":0,"guestNice":0,"stolen":0},{"cpu":"cpu2","user":15030.28,"system":28033.04,"idle":941899.95,"nice":68.08,"iowait":8211.89,"irq":0,"softirq":25.38,"steal":0,"guest":0,"guestNice":0,"stolen":0},{"cpu":"cpu3","user":17497.83,"system":26912.24,"idle":942362.25,"nice":60.36,"iowait":4059.94,"irq":0,"softirq":821.92,"steal":0,"guest":0,"guestNice":0,"stolen":0}],"memory":{"virtual":{"total":16761954304,"available":10951786496,"used":5810167808,"usedPercent":34.66283049473227,"free":474525696,"active":5810085888,"inactive":8564101120,"wired":0,"buffers":998035456,"cached":8517668864,"writeback":0,"dirty":4931584,"writebacktmp":0,"shared":200130560,"slab":1699610624,"pagetables":57794560,"swapcached":839680},"swap":{"total":17043550208,"used":22233088,"free":17021317120,"usedPercent":0.13044869014182328,"sin":1064960,"sout":25591808}},"DISK":{"Usage":{"path":"/","fstype":"ext2/ext3","total":101236436992,"free":57588813824,"used":38481448960,"usedPercent":38.01146119261479,"inodesTotal":6291456,"inodesUsed":1250961,"inodesFree":5040495,"inodesUsedPercent":19.88348960876465}},"HOST_ID":"87c9fbe9-c8c8-4130-a161-04de4ebf52dd"}

**Comment by andybeers** [comment]
ah, i was wondering about the weird capitalization. thumbs up! just updated the model to incorporate the HOST_ID

**Comment by andybeers** [comment]
The naming is starting to scramble me -- why do we call it:
`HOST_ID` on the timeseries JSON, but
`hostid` on the handshake?

**Comment by andybeers** [comment]
Aiight, added a quick n dirty whitelist middleware for the timeseries post route. It's not a great idea, however, since it currently does a database fetch of all handshake-complete servers to authenticate for every single incoming POST (this is bad).

Next TODO is having that db fetch occur once initially, then use in-memory whitelist for further requests.

**Comment by johnshiver** [comment]
ah HOST_ID v. hostid is just what returned by default given the objects i defined, i can change to hostid for everything

**Comment by johnshiver** [comment]
yeah db fetch to do validation for each time series post isnt super efficient, tho not the biggest deal for now. Id suggest adding a TODO and moving on, also note: in-memory cache will need to be refreshed on new server posts :) 

Also TODO: should probably keep all the middleware in a separate directory, not sure if there is a convention for that

**Comment by johnshiver** [comment]
also, what does body-parser package do?

**Comment by andybeers** [comment]
👍

body-parser takes the request JSON body and parses it into JS objects for ya
